bootstrap/0036_sql_tools.sql

```diff
@@ -0,0 +1,74 @@
+
+create or replace procedure admin.validate(obj object, validation_table text, is_create boolean)
+returns text
+language sql
+execute as owner
+as
+$$
+begin
+    -- Select the validation rows. In a create, choose all. In an update, omit rows which are for create_only.
+    let validation_query text := (select tools.templatejs(
+        'select * from internal.validate_{table} where iff({is_create}, true, NOT COALESCE(obj[\'create_only\'], FALSE))',
+        {'table': :validation_table, 'is_create': :is_create}));
+    let rs resultset := (execute immediate :validation_query);
+    let c cursor for rs;
+    for rec in c do
+        begin
+            let q varchar;
+            if (rec['complex']) then
+                -- "complex" validations provide a single row with a "result" column but do so with their own SQL statement.
+                q := rec['sql'];
+            else
+                -- "simple" validations have a condition which is injected in this sql statement
+                q := (select validation.wrap_simple_condition(:rec['sql']));
+            end if;
+
+            let validation_result resultset := (execute immediate :q using (obj));
+            let cc cursor for validation_result;
+            for rr in cc do
+                if (not rr.result) then
+                    return 'Error in validation: ' || rec['message'];
+                end if;
+            end for;
+        exception
+            when other then
+                return 'Error in validation: ' || rec['message'] || ': ' || sqlerrm;
+        end;
+    end for;
+end;
+$$;
+
+create or replace procedure admin.write(obj object, tbl text, key text)
+returns text
+language sql
+execute as owner
+as
+begin
+    let keys array := object_keys(:obj);
+
+    -- The USING portion of the merge statement.
+    let using_select_cols array := (select array_agg(tools.templatejs('obj[\'{col}\'] as {col}', {'col': value})) from table(flatten(input=>:keys)));
```

Comment:
these usages of templatejs seem superflouous. Can we remove them and save on calls to js?